opentelemetry-distro/src/opentelemetry/distro/__init__.py

```diff
@@ -14,143 +14,19 @@
-                for trace_exporter in trace_exporters.split(",")
-            }
-        )
-
-    if EXPORTER_OTLP in exporters:
-        exporters.remove(EXPORTER_OTLP)
-        exporters.add(EXPORTER_OTLP_SPAN)
-
-    return list(exporters)
-
-
-def _init_tracing(
-    exporters: Sequence[SpanExporter], id_generator: IdGenerator
-):
-    # if env var OTEL_RESOURCE_ATTRIBUTES is given, it will read the service_name
-    # from the env variable else defaults to "unknown_service"
-    provider = TracerProvider(
-        id_generator=id_generator(),
-    )
-    trace.set_tracer_provider(provider)
-
-    for _, exporter_class in exporters.items():
-        exporter_args = {}
-        provider.add_span_processor(
-            BatchSpanProcessor(exporter_class(**exporter_args))
-        )
-
-
-def _import_tracer_provider_config_components(
-    selected_components, entry_point_name
-) -> Sequence[Tuple[str, object]]:
-    component_entry_points = {
-        ep.name: ep for ep in iter_entry_points(entry_point_name)
-    }
-    component_impls = []
-    for selected_component in selected_components:
-        entry_point = component_entry_points.get(selected_component, None)
-        if not entry_point:
-            raise RuntimeError(
-                "Requested component '{}' not found in entry points for '{}'".format(
-                    selected_component, entry_point_name
-                )
-            )
-
-        component_impl = entry_point.load()
-        component_impls.append((selected_component, component_impl))
-
-    return component_impls
-
-
-def _import_exporters(
-    exporter_names: Sequence[str],
-) -> Sequence[SpanExporter]:
-    trace_exporters = {}
-
-    for (
-        exporter_name,
-        exporter_impl,
-    ) in _import_tracer_provider_config_components(
-        exporter_names, "opentelemetry_exporter"
-    ):
-        if issubclass(exporter_impl, SpanExporter):
-            trace_exporters[exporter_name] = exporter_impl
-        else:
-            raise RuntimeError(
-                "{0} is not a trace exporter".format(exporter_name)
-            )
-    return trace_exporters
-
-
-def _import_id_generator(id_generator_name: str) -> IdGenerator:
-    # pylint: disable=unbalanced-tuple-unpacking
-    [
-        (id_generator_name, id_generator_impl)
-    ] = _import_tracer_provider_config_components(
-        [id_generator_name.strip()], "opentelemetry_id_generator"
-    )
-
-    if issubclass(id_generator_impl, IdGenerator):
-        return id_generator_impl
-
-    raise RuntimeError("{0} is not an IdGenerator".format(id_generator_name))
-
-
-def _initialize_components():
-    exporter_names = _get_exporter_names()
-    trace_exporters = _import_exporters(exporter_names)
-    id_generator_name = _get_id_generator()
-    id_generator = _import_id_generator(id_generator_name)
-    _init_tracing(trace_exporters, id_generator)
-
-
-class Configurator(BaseConfigurator):
+class OpenTelemetryConfigurator(_OTelSDKConfigurator):
```

Comment:
@owais Replying to your comments...

> I see now that most of the things are private.

This is true! However just a call-out that people who write distros would be subclassing this private `_OTelSDKConfigurator` class as seen in this PR 😕 I think it's a necessary evil and it's okay because there won't be too many distros... but let me know what you think?

Reply:
I think it's acceptable for a while as we try to figure out what this should exactly look like. This shouldn't affect end users so I don't think it's a huge deal.